basic-steps/src/main/java/org/jenkinsci/plugins/workflow/steps/DeleteDirStep.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2014 Jesse Glick.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.steps;
+
+import hudson.Extension;
+import hudson.FilePath;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public final class DeleteDirStep extends AbstractStepImpl {
+
+
+    @DataBoundConstructor 
+    public DeleteDirStep() {
+    }
+
+    @Extension
+    public static final class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(Execution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "deleteDir";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Recursively delete the current directory from the workspace";
+        }
+
+    }
+
+    public static final class Execution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+
+        @StepContextParameter
+        private transient FilePath cwd;
+
+        @Override
+        protected Void run() throws Exception {
+            cwd.deleteRecursive();
+            return null;
```

Comment:
Right, nothing special need be done here. You are thinking of crappy `File.delete`. :-)